win32ss/user/ntuser/window.c

```diff
@@ -596,7 +596,8 @@ LRESULT co_UserFreeWindow(PWND Window,
    /* remove the window already at this point from the thread window list so we
       don't get into trouble when destroying the thread windows while we're still
       in co_UserFreeWindow() */
-   RemoveEntryList(&Window->ThreadListEntry);
+   if (Window->ThreadListEntry.Flink != NULL)
```

Comment:
What would you think about using `InitializeListHead` at the beginning of `IntCreateWindow`, so that `RemoveEntryList` can succeed unconditionally?
The reason I think I'd prefer this is that, if `Flink` becomes NULL for a reason that we don't yet understand, we'll still crash and get closer to understanding said reason, rather than hiding the bug.

Reply:
Imo IntCreateWindow should be refactored so that it doesn't do any callbacks to user mode at all costs. Its role should be to allocate and initialize a valid window. Any failure of callbacks to user mode before it returns will no doubt result to a window in a really messed up state.